creator-node/src/components/healthCheck/healthCheckController.js

```diff
@@ -79,12 +81,44 @@ const healthCheckVerboseController = async (req) => {
   })
 }
 
+/**
+ * Controller for `health_check/fileupload` route
+ * Calls `healthCheckFileUploadService`.
+ */
+const healthCheckFileUploadController = async (req, res, next) => {
+  let { timestamp, randomBytes, signature } = req.query
+  if (!timestamp || !randomBytes || !signature) return errorResponseBadRequest('Missing required query parameters')
+
+  const recoveryObject = { randomBytesToSign: randomBytes, timestamp }
+  const recoveredPublicWallet = recoverWallet(recoveryObject, signature).toLowerCase()
+  const recoveredTimestampDate = new Date(timestamp)
+  const currentTimestampDate = new Date()
+  const requestAge = currentTimestampDate - recoveredTimestampDate
+  if (requestAge >= MAX_HEALTH_CHECK_TIMESTAMP_AGE_MS) {
+    throw new Error(`Submitted timestamp=${recoveredTimestampDate}, current timestamp=${currentTimestampDate}. Maximum age =${MAX_HEALTH_CHECK_TIMESTAMP_AGE_MS}`)
+  }
+  const delegateOwnerWallet = config.get('delegateOwnerWallet').toLowerCase()
+  if (recoveredPublicWallet !== delegateOwnerWallet) {
+    throw new Error("Requester's public key does does not match Creator Node's delegate owner wallet.")
+  }
+
+  try {
+    await promisify(trackFileUpload.single('file'))(req, res)
```

Comment:
we should do some validation to verify the artifacts we expect to exist do exist at this point. we should also remove all the health check artifacts with `removeTrackFolder`

Reply:
As discussed, I think the validation should be handled directly after the upload completes. I will follow up on this in a separate PR.

I've added code to remove the health check artifacts with `removeTrackFolder` and verified that the file gets deleted.